log4j-core/src/main/java/org/apache/logging/log4j/core/pattern/DatePatternConverter.java

```diff
@@ -52,6 +52,10 @@ private abstract static class Formatter {
         public String toPattern() {
             return null;
         }
+
+        public TimeZone getTimeZone() {
+            return null;
+        }
```

Comment:
I would return `TimeZone.getDefault()` here to remove all null checks.

Reply:
Done as suggested